generator_gomod.go

```diff
@@ -54,6 +77,100 @@ func withGomod(g *SourceGenerator, srcSt, worker llb.State, opts ...llb.Constrai
 	}
 }
 
+func withCredHelper(credHelper llb.State, credHelperPath string) RunOptFunc {
+	return func(ei *llb.ExecInfo) {
+		llb.AddMount(credHelperPath, credHelper, llb.SourcePath(GitCredentialHelperGomod)).SetRunOption(ei)
+	}
+}
+
+func (g *SourceGenerator) mountGitAuthConfig(mountPoint, basename string) llb.RunOption {
+	return RunOptFunc(func(ei *llb.ExecInfo) {
+		if g.Gomod == nil || g.Gomod.Auth == nil {
+			return
+		}
+
+		var b bytes.Buffer
+		enc := gob.NewEncoder(&b)
+		if err := enc.Encode(g.Gomod.Auth); err != nil {
+			panic("cannot marshal dalec spec bytes")
+		}
+
+		st := llb.Scratch().File(llb.Mkfile("/"+basename, 0o644, b.Bytes()))
+		llb.AddMount(mountPoint, st).SetRunOption(ei)
+	})
+}
+
+func (g *SourceGenerator) gitconfigGeneratorScript(scriptPath, configPath string) llb.State {
+	var script bytes.Buffer
+
+	sortedHosts := SortMapKeys(g.Gomod.Auth)
+	if len(sortedHosts) > 0 {
+		fmt.Fprintln(&script, `set -eu`)
+	}
+
+	for _, host := range sortedHosts {
+		if sshConfig := g.Gomod.Auth[host].SSH; sshConfig != nil {
+			username := "git"
+			if sshConfig.Username != "" {
+				username = sshConfig.Username
+			}
+
+			fmt.Fprintf(&script, `git config --global "url.ssh://%[1]s@%[2]s/.insteadOf" https://%[2]s/`, username, host)
+			script.WriteRune('\n')
+			continue
+		}
+
+		fmt.Fprintf(&script, `git config --global credential."https://%s".helper "gomod %s"`, host, configPath)
```

Comment:
Let's give this a name that's more dalec specific?
`dalec-git-credentials`?

Reply:
You can give it an absolute path also, so I'm just giving it the path to the frontend.

Reply:
done.